js/create_progress_bar.js

```diff
@@ -0,0 +1,196 @@
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+/**
+ * @param     parameters
+ * @param     {HTMLElement} parameters.container Mandatory. The progress bar's unique key.
+ * @param     {string} parameters.key Mandatory. The progress bar's unique key.
+ * @param     {null|function} parameters.progress_callback The function that will be called for each progress response. If the return value is "false", this stops the progress checks.
+ * @param     {null|function} parameters.error_callback The function that will be called for each error, either exceptions or non-200 HTTP responses. Stops the progress checks by default, unless you return a true-ish value from the callback, or unless the error is non-recoverable and implies stopping
+ * @return    {{start: function, stop: function}}
+ */
+function create_progress_bar(parameters)
+{
+    if (!parameters.key) {
+        throw new Error('Progress key is mandatory.');
+    }
+    if (!parameters.container) {
+        throw new Error('Progress container is mandatory.');
+    }
+    if (!(parameters.container instanceof HTMLElement)) {
+        throw new Error(`Progress key must be an HTML element, "${parameters.container?.constructor?.name || typeof parameters.container}" found.`);
+    }
+
+    const main_container = document.createElement('div');
+    main_container.innerHTML = `
+        <div style="padding-left: 20px; padding-right: 20px;">
+            <div class="progress" style="height: 15px;">
+                <div class="progress-bar bg-info" role="progressbar" style="width:0;" aria-valuenow="0" aria-valuemin="0" aria-valuemax="100">0%</div>
+            </div>
+            <div class="messages_container"></div>
+        </div>
+    `;
+    const progress_bar = main_container.querySelector('.progress-bar');
+    const messages_container = main_container.querySelector('.messages_container');
+
+    parameters.container.appendChild(main_container);
+
+    const start_timeout = 250;
+
+    let is_running = true;
+    let abort_controller = new AbortController();
+
+    function escapeHtml(text) {
+        const map = {
+            '&': '&amp;',
+            '<': '&lt;',
+            '>': '&gt;',
+            '"': '&quot;',
+            "'": '&#039;'
+        };
+
+        return text.replace(/[&<>"']/g, (m) => map[m]);
+    }
```

Comment:
You should have access to lodash functions here (try `_.escape()`), it is loaded in the global scope.